src/connector/aiConnector.ts

```diff
@@ -0,0 +1,42 @@
+import connectorsManager from 'fontoxml-configuration/src/connectorsManager';
+import RequestData from 'fontoxml-connector/src/RequestData';
+
+import type { AiGenerateResponse, AiHistory } from '../types';
+
+class AiConnector {
+	/**
+	 * Generates an AI response, using the POST /generate CMS endpoint.
+	 *
+	 * @param prompt The prompt.
+	 * @param history History of all the previous prompts in this session.
+	 * @param context The selection we have in the editor.
+	 */
+	public async generate(
+		prompt: string,
+		history: AiHistory[],
+		context?: string
+	): Promise<AiGenerateResponse> {
+		const requestData = new RequestData();
+
+		requestData.setContentType('json');
+		requestData.addPostParameters({
+			context: connectorsManager.getCmsClient().createContext(),
+			prompt,
+			contextContent: context,
+			history,
+		});
+
+		const result = await connectorsManager
+			.getCmsClient()
+			.sendRequest('POST', `/ai-assistant/generate`, requestData)
```

Comment:
Should we consider validating the response, like we normally do in the platform?

Reply:
You'll need private API for that. I'll have to look into an API proposal for that.